src/main/java/com/gooddata/md/MetadataService.java

```diff
@@ -97,6 +101,30 @@ public MetadataService(RestTemplate restTemplate) {
     }
 
     /**
+     * Retrieves a list of objects corresponding to the supplied list of URIs.
+     *
+     * @param project project that contains the objects to be retrieved
+     * @param uris list of URIs
+     * @return list of metadata objects corresponding to the supplied URIs
+     */
+    public List<Obj> getObjsByUris(Project project, List<String> uris) {
```

Comment:
isn't this list paged?

Reply:
no, but there is a limit on how many objects you can fetch. right now I leave it up to the user to comply with the limit. gdc-backend provides more functionality (sheilds user from this by splitting it to multiple requests based on how many uris was provided) - I have no problem implementing it like that even here, but thought I would rather keep it simple and leave it up to the client as then we may run into other things like should we run these individual requests in parallel or sequentially, etc. let me know if you prefer the more complex approach (automated splitting into several requests, 50 uris per request).

Reply:
sure, keep it simple, definitely no magic!

what happens if the limit is exceed?

Reply:
4xx status